test/functional/wallet_pruning.py

```diff
@@ -0,0 +1,148 @@
+        height = int(best_block["height"]) + 1
+        self.nTime = max(self.nTime, int(best_block["time"])) + 1
+        previousblockhash = int(best_block["hash"], 16)
+        big_script = CScript([OP_RETURN] + [OP_TRUE] * 950000)
+        for _ in range(n):
+            block = create_block(hashprev=previousblockhash, ntime=self.nTime, coinbase=create_coinbase(height, script_pubkey=big_script))
+            block.solve()
+
+            # Submit to the node
+            node.submitblock(block.serialize().hex())
+
+            previousblockhash = block.sha256
+            height += 1
+
+            # Simulate 10 minutes of work time per block
+            # Important for matching a timestamp with a block +- some window
+            self.nTime += 600
+            for n in self.nodes:
+                if n.running:
+                    n.setmocktime(self.nTime) # Update node's time to accept future blocks
+
+
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [
+            [], # node dedicated to mining
+            ['-prune=550'], # node dedicated to testing pruning
+        ]
+
+    def setup_nodes(self):
+        self.add_nodes(self.num_nodes, extra_args=self.extra_args)
+        self.start_node(0)
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def test_wallet_import_pruned_test(self):
+        self.log.info("Make sure we can import wallet when pruned and required blocks are still available")
+
+        wname = "wallet_pruned.dat"
+
+        # Export wallet from node 0
+        self.nodes[0].createwallet(wallet_name="wallet_pruned", descriptors=False, load_on_startup=True)
+        self.nodes[0].dumpwallet(os.path.join(self.nodes[0].datadir, wname))
+
+        # Add mature coins to the wallet
+        self.generatetoaddress(self.nodes[0], 1, self.nodes[0].getnewaddress(), sync_fun=lambda: None)
+        self.generate(self.nodes[0], 100)
+
+        # Import wallet into node 1
+        self.nodes[1].createwallet(wallet_name="wallet_pruned", descriptors=False, load_on_startup=True)
+        self.nodes[1].importwallet(os.path.join(self.nodes[0].datadir, wname))
+
+        # Make sure the wallets have the same balance
+        assert_equal(self.nodes[1].getbalance(), self.nodes[0].getbalance())
+
+        # Mine some blocks, pruning should not have removed the blocks required for the wallet
+        self.mine_large_blocks(self.nodes[0], 5)
+        self.sync_all()
+
+        # Import wallet, should still work
+        self.nodes[1].importwallet(os.path.join(self.nodes[0].datadir, wname))
+        self.log.info("Wallet successfully imported on pruned node")
+
+    def test_wallet_import_pruned_with_missing_blocks(self):
+        self.log.info("Make sure we cannot import wallet when pruned and required blocks are not available")
+
+        wname = "wallet_init.dat"
```

Comment:
In c46f5fe5:

Tiny nit, this could be passed as an argument so we avoid hardcoding file names more than once.